util/webhook/webhook.go

```diff
@@ -244,14 +271,66 @@ func affectedRevisionInfo(payloadIf interface{}) (webURLs []string, revision str
 	return webURLs, revision, change, touchedHead, changedFiles
 }
 
+func fetchDiffstatFromBitbucket(workspace, repoSlug, spec, accessToken string) ([]string, error) {
```

Comment:
I'm concerned about this. iirc our webhook handler isn't auth-protected. So anyone with API access could hammer the endpoint and trigger SSRF. I don't have time to give a thorough review right now, but requesting changes so this doesn't get merged without further conversation.